plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -100,6 +110,65 @@ const Extensions = ( {
 	);
 };
 
+const getAllowTrackingOption = async () =>
+	resolveSelect( OPTIONS_STORE_NAME ).getOption(
+		'woocommerce_allow_tracking'
+	);
+
+const handleTrackingOption = assign( {
+	optInDataSharing: (
+		_context,
+		event: DoneInvokeEvent< 'no' | 'yes' | undefined >
+	) => event.data !== 'no',
+} );
+
+const recordTracksIntroCompleted = () => {
+	recordEvent( 'storeprofiler_step_complete', {
+		step: 'store_details',
+		wc_version: getSetting( 'wcVersion' ),
+	} );
+};
+
+const recordTracksIntroSkipped = () => {
+	recordEvent( 'storeprofiler_store_details_skip' );
+};
+
+const recordTracksIntroViewed = () => {
+	recordEvent( 'storeprofiler_step_view', {
+		step: 'store_details',
+		wc_version: getSetting( 'wcVersion' ),
+	} );
+};
+
+const updateTrackingOption = (
+	_context: CoreProfilerStateMachineContext,
+	event: IntroOptInEvent
+) => {
+	if (
+		event.payload.optInDataSharing &&
+		typeof window.wcTracks.enable === 'function'
+	) {
+		window.wcTracks.enable( () => {
+			initializeExPlat();
+		} );
+	} else if ( ! event.payload.optInDataSharing ) {
+		window.wcTracks.isEnabled = false;
+	}
+
+	const trackingValue = event.payload.optInDataSharing ? 'yes' : 'no';
```

Comment:
If it is easier we can just use 'yes' 'no' in the context, it slipped my mind that it's a stringly value and not boolean when I was scaffolding 😅

Reply:
I think it’s fine either way because if it's a stringly value, we still need to convert it to boolean for checkbox. 😅